web-api/src/business/useCaseHelper/trialSessions/trialSessionCalendaring/writeTrialSessionDataToExcel.ts

```diff
@@ -0,0 +1,355 @@
+import { SESSION_TYPES } from '@shared/business/entities/EntityConstants';
+import ExcelJS from 'exceljs';
+
+type ColumnObject = { header: string; key: string; width?: number };
+
+const CITY_TITLE_CELL_LOCATION = 'A2';
+
+export const writeTrialSessionDataToExcel = async ({
+  caseCountsAndSessionsByCity,
+  incorrectSizeRegularCases,
+  userMessages,
+  weeks,
+}: {
+  weeks: string[];
+  caseCountsAndSessionsByCity: CaseCountsAndSessionsByCity;
+  incorrectSizeRegularCases: EligibleCase[];
+  userMessages: string[];
+}) => {
+  const workbook = new ExcelJS.Workbook();
+  const worksheetOptions = { properties: { outlineLevelCol: 2 } };
+  const worksheet = workbook.addWorksheet(
+    'Suggested Session Calendar',
+    worksheetOptions,
+  );
+
+  const rowsByCity = getRowsByCity({ caseCountsAndSessionsByCity, weeks });
+
+  worksheet.columns = getColumns({ weeks });
+
+  for (const cityStateString in rowsByCity) {
+    const populatedRow = populateRow({
+      caseCountsAndSessionsByCity,
+      cityStateString,
+      row: rowsByCity[cityStateString],
+    });
+    worksheet.addRow(populatedRow);
+  }
+
+  worksheet.eachRow(row => {
+    row.eachCell({ includeEmpty: true }, cell => {
+      const { alignment, border, fill, font } = getCellStyle(cell.value);
+      cell.alignment = alignment;
+      cell.border = border;
+      cell.fill = fill;
+      cell.font = font;
+    });
+  });
+
+  worksheet.insertRow(1, [null, 'Week Of']);
+
+  const emptyCounterRow = weeks.reduce((acc, week) => {
+    acc[week] = 0;
+    return acc;
+  }, {});
+
+  const counterRow = worksheet.addRow({
+    city: 'No. of Sessions',
+    ...emptyCounterRow,
+  });
+
+  const countColumnLength = Object.keys(rowsByCity).length; // number of cells in a column that we care about
+
+  counterRow.eachCell(cell => {
+    const cellLetter = cell.$col$row.split('$')[1];
+    const { alignment, border, value } = getCounterCellData(
+      cellLetter,
+      countColumnLength,
+    );
+    cell.alignment = alignment;
+    cell.border = border;
+    cell.value = value;
+  });
+
+  const cityTitleCell = worksheet.getCell(CITY_TITLE_CELL_LOCATION);
+
+  cityTitleCell.border = {
+    bottom: undefined,
+    left: undefined,
+    right: undefined,
+    top: undefined,
+  };
+
+  cityTitleCell.fill = {
+    pattern: 'none',
+    type: 'pattern',
+  };
+
+  if (incorrectSizeRegularCases.length > 0) {
+    const incorrectlySizedCasesTab = workbook.addWorksheet(
+      'Incorrectly Sized Cases',
+    );
+
+    incorrectlySizedCasesTab.columns = [
+      {
+        header: 'City',
+        key: 'city',
+      },
+      { header: 'Docket Numbers' },
+    ];
+
+    getIncorrectlySizedCasesRows(incorrectSizeRegularCases).forEach(row => {
+      incorrectlySizedCasesTab.addRow(row);
+    });
+  }
+
+  if (userMessages.length > 0) {
+    const warningsTab = workbook.addWorksheet('Warnings', {
+      properties: { tabColor: { argb: 'ffb50909' } },
```

Comment:
Minor, but is it worth pulling this out as a constant?

Reply:
You've talked me into it